File: 0020-valid-parentheses/0020-valid-parentheses.java
class Solution {
    public boolean isValid(String s) {
        Stack<Character> stack = new Stack<>();
        char[] charArr = s.toCharArray();
        for(char ch : charArr){
            if(ch == '(' || ch == '{' || ch == '['){
                stack.push(ch);
            }
            else{
                if(stack.size()<=0){
                    return false;
                }
                char temp = stack.pop();
                if((temp == '(' && ch != ')') || (temp == '{' && ch != '}') || (temp == '[' && ch != ']')){
                    return false;
                }
            }
        }
        if(stack.size()==0){
            return true;
        }
        return false;
    }
}

File: 0020-valid-parentheses/0020-valid-parentheses.py
class Solution:
    def isValid(self, s: str) -> bool:
        bracketMap = { '(': ')', '[': ']', '{': '}' }
        stack = []
        for ch in s:
            if ch in bracketMap:
                stack.append(ch)
            else:
                if stack and ch == bracketMap[stack[-1]]:
                    stack.pop()
                else:
                    return False
        return True if not stack else False 